subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/ResolveConfigurationResolutionBuildOperationResult.java

```diff
@@ -18,37 +18,27 @@
 
 import com.google.common.collect.Maps;
 import org.gradle.api.Action;
+import org.gradle.api.artifacts.result.ResolutionResult;
 import org.gradle.api.artifacts.result.ResolvedComponentResult;
 import org.gradle.api.internal.artifacts.result.ResolvedComponentResultInternal;
-import org.gradle.internal.Actions;
 import org.gradle.internal.operations.trace.CustomOperationTraceSerialization;
 
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
 
 class ResolveConfigurationResolutionBuildOperationResult implements ResolveConfigurationDependenciesBuildOperationType.Result, CustomOperationTraceSerialization {
```

Comment:
@facewindu @bigdaz can you sanity check this part of the change? I don't believe this changes the behavior visible to the build scan plugin.

We added checks to `DefaultConfiguration` to make sure the thread resolving the configuration held the project lock for the configuration.  This build operation was the source of some failures because getting a new resolution result would cause configurations to be re-resolved without the project lock.  I think we didn't notice anything before because the re-resolve was essentially a no-op because this build operation is created after we've resolved the configuration.

Reply:
cc @ldaley